src/Lecture3.hs

```diff
@@ -134,10 +143,17 @@ data List1 a = List1 a [a]
 
 -- | This should be list append.
 instance Semigroup (List1 a) where
-
+  (<>) :: List1 a -> List1 a -> List1 a
+  List1 fa ra <> List1 fb rb = List1 fa (ra ++ [fb] ++ rb)
```

Comment:
You can slightly improve the performance here by using the `:` operator instead of list append `<>` or `++` 🐎 
```suggestion
  List1 fa ra <> List1 fb rb = List1 fa (ra ++ fb : rb)
```